### ISSUE TITLE
local insecure registry not working

### ISSUE THREAD

**Issue by tmbdev** [canonical/microk8s#3288]
TL;DR I'm running a registry under podman now and have gotten this to work with that registry. I'm still not sure what the original problem was, but I consider the bug closed.

#### Summary

I'm trying to set up a local, insecure registry as described here: https://microk8s.io/docs/registry-private

I have started the registry and I can push images with podman (using an http configuration). However microk8s still uses HTTPS. That is, I can't "microk8s.kubectl apply -f mypod.yml" with an image referencing "nix:32000/vadd" or "nix:32000/vadd:latest". This also fails with the "microk8s ctr" command, which is easier to show and debug.

I don't know what the cause is, and I don't even know how to debug this any further. The most detailed debugging output I can get is the one shown below.

#### What Happened?

```bash
$ cat /var/snap/microk8s/current/args/certs.d/nix:32000/hosts.toml
server = "http://nix:32000"

[host."nix:32000"]
capabilities = ["pull", "resolve"]
$ microk8s ctr --debug image pull nix:32000/vadd:latest
DEBU[0000] fetching                                      image="nix:32000/vadd:latest"
DEBU[0000] resolving                                     host="nix:32000"
DEBU[0000] do request                                    host="nix:32000" request.header.accept="application/vnd.docker.distribution.manifest.v2+json, application/vnd.docker.distribution.manifest.list.v2+json, application/vnd.oci.image.manifest.v1+json, application/vnd.oci.image.index.v1+json, */*" request.header.user-agent=containerd/v1.5.13 request.method=HEAD url="https://nix:32000/v2/vadd/manifests/latest"
INFO[0000] trying next host                              error="failed to do request: Head \"https://nix:32000/v2/vadd/manifests/latest\": http: server gave HTTP response to HTTPS client" host="nix:32000"
ctr: failed to resolve reference "nix:32000/vadd:latest": failed to do request: Head "https://nix:32000/v2/vadd/manifests/latest": http: server gave HTTP response to HTTPS client
$ snap info microk8s
...
snap-id:      EaXqgt1lyCaxKaQCU349mlodBkDCXRcg
tracking:     1.24/edge
refresh-date: today at 10:45 PDT
channels:
...
$ 
```

#### What Should Happen Instead?

Given the .toml configuration file, microk8s should have used HTTP to access the local, private, insecure registry.

#### Reproduction Steps

(See above.)

#### Introspection Report

[inspection-report-20220630_233235.tar.gz](https://github.com/canonical/microk8s/files/9026667/inspection-report-20220630_233235.tar.gz)

#### Can you suggest a fix?

no, sorry

#### Are you interested in contributing with a fix?

sorry, but I'm not familiar enough with the inner workings of microk8s

**Comment by neoaggelos** [comment]
Hi @tmbdev,

I see that you have closed the issue. I went over the docs at https://microk8s.io/docs/registry-private and everything should be working. If you had any issues with the instructions, please let us know.

Thank you!

**Comment by neoaggelos** [comment]
Hi @tmbdev 

In the inspection report for this, I see the following config:
```toml
server = "http://nix:32000"

[host."nix:32000"]
capabilities = ["pull", "resolve"]
```

It should instead be (as shown in the documentation page):

```toml
server = "http://nix:32000"

[host."http://nix:32000"]
capabilities = ["pull", "resolve"]
```

containerd will then try to talk to your registry via http

**Comment by tmbdev** [comment]
Thanks.

**Comment by jcabrerazuniga** [comment]
If using Self Signed Cert:

1. Trust the certificate authority (recommended)
Identify the CA: If the certificate is self-signed, it's its own CA. If it's signed by an internal CA, you need that CA's certificate.
Add the CA to your system's trust store: On most Linux distributions, you can add the CA certificate to /usr/local/share/ca-certificates/ and then run sudo update-ca-certificates to incorporate it into the system's trust store.

Then the provided instructions worked for me
